public/components/datasources/components/new/configure_datasource.tsx

```diff
@@ -0,0 +1,211 @@
+            selectedQueryPermissionRoles={selectedQueryPermissionRoles}
+            setSelectedQueryPermissionRoles={setSelectedQueryPermissionRoles}
+          />
+        );
+      default:
+        return <></>;
+    }
+  };
+
+  const ReviewDatasourceConfiguration = (configurationProps: { datasourceType: string }) => {
+    const { datasourceType } = configurationProps;
+    switch (datasourceType) {
+      case 'S3':
+        return (
+          <ReviewS3Datasource
+            currentName={name}
+            currentDetails={details}
+            currentArn={arn}
+            currentStore={store}
+            selectedQueryPermissionRoles={selectedQueryPermissionRoles}
+          />
+        );
+      default:
+        return <></>;
+    }
+  };
+
+  const ReviewSaveOrCancel = useCallback(() => {
+    return (
+      <EuiBottomBar>
+        <EuiFlexGroup justifyContent="flexEnd">
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty
+              onClick={() => {
+                window.location.hash = '#/new';
+              }}
+              color="ghost"
+              size="s"
+              iconType="cross"
+            >
+              Cancel
+            </EuiButtonEmpty>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiButton
+              onClick={page === 'review' ? () => setPage('configure') : () => {}}
+              color="ghost"
+              size="s"
+              iconType="arrowLeft"
+            >
+              Previous
+            </EuiButton>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiButton
+              onClick={
+                page === 'review'
+                  ? () => createDatasource()
+                  : () => {
+                      setPage('review');
+                    }
+              }
+              size="s"
+              iconType="arrowRight"
+              fill
+            >
+              {page === 'configure' ? `Review Configuration` : `Connect to ${type}`}
+            </EuiButton>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </EuiBottomBar>
+    );
+  }, [page]);
+
+  const createDatasource = () => {
+    http!
+      .post(`${DATACONNECTIONS_BASE}`, {
+        body: JSON.stringify({
+          name,
+          allowedRoles: selectedQueryPermissionRoles.map((role) => role.label),
+          connector: 's3glue',
+          properties: {
+            'glue.auth.type': 'iam_role',
+            'glue.auth.role_arn': arn,
+            'glue.indexstore.opensearch.uri': store,
+            'glue.indexstore.opensearch.auth': false,
+            'glue.indexstore.opensearch.region': 'us-west-2',
+          },
+        }),
+      })
+      .then(() => {
+        setToast(`Data source ${name} created successfully!`);
+        window.location.hash = '#/manage';
+      })
+      .catch((err) => {
+        setToast(`Data source ${name} created successfully!`);
+        window.location.hash = '#/manage';
```

Comment:
These are doing the same thing temporarily, since backend is returning 500 even when successfully created

Reply:
When this PR is merged - https://github.com/opensearch-project/sql/pull/2156, we can separate these out properly